apps/prairielearn/src/ee/pages/lti13CourseNavigation/lti13CourseNavigation.ts

```diff
@@ -1,10 +1,182 @@
 import { Router } from 'express';
+import asyncHandler from 'express-async-handler';
 
+import * as error from '@prairielearn/error';
+import {
+  loadSqlEquiv,
+  queryOptionalRow,
+  queryAsync,
+  queryOneRowAsync,
+} from '@prairielearn/postgres';
+
+import { Lti13CourseInstanceSchema } from '../../../lib/db-types.js';
+import {
+  selectCourseInstanceById,
+  selectCourseInstancesWithStaffAccess,
+  CourseInstanceAuthz,
+} from '../../../models/course-instances.js';
+import { selectCoursesWithEditAccess } from '../../../models/course.js';
+import { Lti13Claim } from '../../lib/lti13.js';
+
+import {
+  Lti13CourseNavigationInstructor,
+  Lti13CourseNavigationNotReady,
+  Lti13CourseNavigationDone,
+} from './lti13CourseNavigation.html.js';
+
+const sql = loadSqlEquiv(import.meta.url);
 const router = Router({ mergeParams: true });
 
-// Placeholder to be enlarged later
-router.all('/', (_req, res) => {
-  res.redirect('/pl');
-});
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    if ('done' in req.query) {
+      res.send(
+        Lti13CourseNavigationDone({
+          resLocals: res.locals,
+          lti13_instance_id: req.params.lti13_instance_id,
+        }),
+      );
+      return;
+    }
+
+    const ltiClaim = new Lti13Claim(req);
+    const courseName = `${ltiClaim.context?.label}: ${ltiClaim.context?.title}`;
```

Comment:
Can we be smarter here so that we don't end up with `CS 101: null` or `undefined: Potions` or the like?